src/main/java/com/uber/jenkins/phabricator/BuildResultProcessor.java

```diff
@@ -85,7 +85,11 @@ public BuildResultProcessor(
     }
 
     public Result getBuildResult() {
-        return this.build.getResult();
+        if (this.build.getResult() == null) {
```

Comment:
Can you explain why we would want to return success on null? It seems a bit counterintuitive. Are there any api docs defining this new behavior?

Reply:
on oldschool freestyle jobs after the build reached the last build step it set the build result to success, but in pipeline there is no definite end of build blocks, so it won't explicitly set SUCCESS, hence the previous workaround, where it was needed to run this before the notifier:
```groovy
currentBuild.result = currentBuild.result ?: 'SUCCESS'
```

Reply:
null == success so far. IMHO it was a genuine missing null check in the plugin code. In freestyle it is always set and in pipeline it is always considered SUCCESS.

Reply:
makes sense